tlslite/keyexchange.py

```diff
@@ -497,10 +499,22 @@ def makeServerKeyExchange(self, sigHash=None):
         self.group_id = getFirstMatching(client_curves, self.acceptedCurves)
         if self.group_id is None:
             raise TLSInsufficientSecurity("No mutual groups")
-        generator = getCurveByName(GroupName.toRepr(self.group_id)).generator
-        self.ecdhXs = ecdsa.util.randrange(generator.order())
+        if self.group_id in [GroupName.x25519, GroupName.x448]:
+            if self.group_id == GroupName.x25519:
+                generator = bytearray(X25519_G)
+                fun = x25519
+                self.ecdhXs = getRandomBytes(divceil(X25519_ORDER, 8))
```

Comment:
it's the same order as defined in [section 3](https://tools.ietf.org/html/rfc7748#section-3) with the value taken from [section 4.1](https://tools.ietf.org/html/rfc7748#section-4.1). I think I may be missing what's confusing about it...